lib/src/date-input/DateInput.tsx

```diff
@@ -278,46 +151,37 @@ const DxcDateInput = React.forwardRef<RefType, DateInputPropsType>(
                 margin={margin}
                 size={size}
                 tabIndex={tabIndex}
-                ref={refDate}
+                ref={dateRef}
               />
-              <Popover
-                onKeyDown={handleCalendarOnKeyDown}
-                open={isOpen}
-                anchorEl={anchorEl}
-                anchorOrigin={{
-                  vertical: "bottom",
-                  horizontal: "left",
-                }}
-                transformOrigin={{
-                  vertical: "top",
-                  horizontal: "center",
-                }}
-                PaperProps={{
-                  style: {
-                    marginTop: "10px",
-                  },
-                }}
-              >
-                <ClickAwayListener onClickAway={closeCalendar}>
-                  <Paper role="dialog" aria-modal="true">
-                    <DatePicker
-                      variant="static"
-                      value={getValueForPicker(value ?? innerValue, format)}
-                      onChange={(date) => handleCalendarOnClick(date)}
-                      format={format}
-                      disabled={disabled}
-                    />
-                  </Paper>
-                </ClickAwayListener>
-              </Popover>
-            </StyledDPicker>
-          </MuiPickersUtilsProvider>
-        </MuiThemeProvider>
+            </Popover.Trigger>
+            <StyledContent
+              sideOffset={error ? -18 : 2}
+              align="end"
+              aria-modal={true}
+              onFocusOutside={handleFocusOutside}
```

Comment:
For some reason, the Radix Popover method onFocusOutside is never being called (you can check it by simply putting a console.log). We can fix this by replacing it with the `onBlur` event like this:

```
    const handleDatePickerOnBlur = (event) => {
      if (!event?.currentTarget.contains(event.relatedTarget)) {
        console.log("blur")
        closeCalendar();
      }
    };
```

This will allow us to remove also the window listener `window.addEventListener("blur", closeCalendar);`. This is a more specific and correct answer.

Reply:
Done!